src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -1369,3 +1401,375 @@ def repack_with_tag(tag: str,
+    DROP_TABLE_TAG = HB_SUBSET_SETS_DROP_TABLE_TAG
+    NAME_ID = HB_SUBSET_SETS_NAME_ID
+    NAME_LANG_ID = HB_SUBSET_SETS_NAME_LANG_ID
+    LAYOUT_FEATURE_TAG = HB_SUBSET_SETS_LAYOUT_FEATURE_TAG
+    LAYOUT_SCRIPT_TAG = HB_SUBSET_SETS_LAYOUT_SCRIPT_TAG
+
+
+cdef class SubsetInput:
+    cdef hb_subset_input_t* _input
+
+    def __cinit__(self):
+        self._input = hb_subset_input_create_or_fail()
+        if self._input is NULL:
+            raise MemoryError()
+
+    def __dealloc__(self):
+        if self._input is not NULL:
+            hb_subset_input_destroy(self._input)
+
+    def subset(self, source: Face) -> Face:
+        new_face = hb_subset_or_fail(source._hb_face, self._input)
+        if new_face == NULL:
+            raise RuntimeError("Subsetting failed")
+        return Face.from_ptr(new_face)
+
+    def keep_everything(self):
+        hb_subset_input_keep_everything(self._input)
+
+    def pin_axis_to_default(self, face: Face, tag: str) -> bool:
+        hb_tag = hb_tag_from_string(tag.encode("ascii"), -1)
+        return hb_subset_input_pin_axis_to_default(
+            self._input, face._hb_face, hb_tag
+        )
+
+    def pin_axis_location(self, face: Face, tag: str, value: float) -> bool:
+        hb_tag = hb_tag_from_string(tag.encode("ascii"), -1)
+        return hb_subset_input_pin_axis_location(
+            self._input, face._hb_face, hb_tag, value
+        )
+
+    @property
+    def unicode_set(self) -> Set:
+        return Set.from_ptr(hb_set_reference (hb_subset_input_unicode_set(self._input)))
+
+    @property
+    def glyph_set(self) -> Set:
+        return Set.from_ptr(hb_set_reference (hb_subset_input_glyph_set(self._input)))
+
+    def sets(self, set_type : SubsetInputSets) -> Set:
+        return Set.from_ptr(hb_set_reference (hb_subset_input_set(self._input, set_type)))
+
+    def __getitem__(self, set_type : SubsetInputSets) -> Set:
+        return self.sets(set_type)
+
+    def __setitem__(self, set_type : SubsetInputSets, iterable) -> Set:
+        s = self.sets(set_type)
+        s.set(iterable)
+
+    @property
+    def flags(self) -> SubsetFlags:
+        cdef unsigned subset_flags = hb_subset_input_get_flags(self._input)
+        return SubsetFlags(subset_flags)
+
+    @flags.setter
+    def flags(self, flags: SubsetFlags) -> None:
+        hb_subset_input_set_flags(self._input, int(flags))
+
+
+class SubsetFlags(IntFlag):
+    DEFAULT = HB_SUBSET_FLAGS_DEFAULT
+    NO_HINTING = HB_SUBSET_FLAGS_NO_HINTING
+    RETAIN_GIDS = HB_SUBSET_FLAGS_RETAIN_GIDS
+    DESUBROUTINIZE = HB_SUBSET_FLAGS_DESUBROUTINIZE
+    NAME_LEGACY = HB_SUBSET_FLAGS_NAME_LEGACY
+    SET_OVERLAPS_FLAG = HB_SUBSET_FLAGS_SET_OVERLAPS_FLAG
+    PASSTHROUGH_UNRECOGNIZED = HB_SUBSET_FLAGS_PASSTHROUGH_UNRECOGNIZED
+    NOTDEF_OUTLINE = HB_SUBSET_FLAGS_NOTDEF_OUTLINE
+    GLYPH_NAMES = HB_SUBSET_FLAGS_GLYPH_NAMES
+    NO_PRUNE_UNICODE_RANGES = HB_SUBSET_FLAGS_NO_PRUNE_UNICODE_RANGES
```

Comment:
Can we move this up next to `SubsetInputSets`, it looks odd with one before the `SubsetInput` class and one after it.